dicodile/utils/tests/test_csc.py

```diff
@@ -30,3 +32,21 @@ def test_ztz(valid_support, atom_support, sparsity):
     X_hat = reconstruct(z, D)
 
     assert np.isclose(cost, np.dot(X_hat.ravel(), X_hat.ravel()))
+
+
+def test_dense_convolve_multi_uv_shape():
+
+    n_channels = 3
+    sig_shape = (n_channels, 800, 600)
+    atom_shape = (n_channels, 40, 30)
+    atom_support = atom_shape[1:]
+    n_atoms = 25
+    valid_support = get_valid_support(sig_support=sig_shape[1:],
+                                      atom_support=atom_support)
+
+    z_hat = np.ones((n_atoms, *valid_support))
+    u = np.ones((n_atoms, n_channels))
+    v = np.ones((n_atoms, *atom_support))
+    Xi = _dense_convolve_multi_uv(z_hat, (u, v))
+
+    assert Xi.shape == sig_shape
```

Comment:
Could you also add a test to check that the output of `_dense_convolve_multi{_uv,_d}` match?

Reply:
Good idea, by the way, should we change `get_D` to take a `(u,v)` tuple?

Reply:
yes, thanks for the troubleshooting with `get_D`